### PR TITLE
Jedwards/standard makefile args

### PR DESCRIPTION
**Pull request by jedwards4b**
Cleanup of build.  

Test suite: scripts_regression_tests.py nuopc build suite 
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes 

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

File: scripts/Tools/Makefile [hunk 1]
```diff
@@ -69,12 +69,12 @@ endif
...
    USE_CXX = true
 endif
-ifeq ($(strip $(USE_KOKKOS)), TRUE)
-   USE_CXX = true
-endif
-
-ifeq ($(strip $(USE_FMS)), TRUE)
-  SLIBS += -lfms
+ifeq ($(CIME_MODEL),e3sm)
+  HOMME_TARGET ?= $(shell $(CASEROOT)/xmlquery --caseroot $(CASEROOT) CAM_TARGET --value)
```
Comment (jgfouca) [inline scripts/Tools/Makefile]: Why was this change made? It looks like it adds-back HOMME checks that I removed.

File: scripts/lib/CIME/build.py [hunk 2]
```diff
@@ -9,6 +9,29 @@
 
 logger = logging.getLogger(__name__)
 
+def get_standard_makefile_args(case):
+    variables = ["CASEROOT", "CASETOOLS", "CIMEROOT", "COMP_INTERFACE",
+                 "COMPILER", "DEBUG", "EXEROOT", "INCROOT", "LIBROOT",
+                 "MACH", "MPILIB", "NINST_VALUE", "OS", "PIO_VERSION",
+                 "SHAREDLIBROOT", "SMP_PRESENT", "USE_ESMF_LIB", "USE_MOAB",
```
Comment (jgfouca) [inline scripts/lib/CIME/build.py]: Need USE_KOKKOS here, at least for e3sm.

**Comment by jgfouca** [comment]
Looks good except for USE_KOKKOS stuff. We can either make it a case setting for e3sm or you can add it by hand to makefile args.

File: scripts/lib/CIME/build.py (hunk 2 above)
Comment (jedwards4b) [reply scripts/lib/CIME/build.py]: I put it in line 497 of this file, still as an environment variable - to be consistent it should be a case variable as are USE_TRILINOS and USE_ALBANY at line 467+

File: scripts/Tools/Makefile (hunk 1 above)
Comment (jedwards4b) [reply scripts/Tools/Makefile]: i'll remove it again - it's hard to change stuff you can't test...

**Comment by jedwards4b** [comment]
@jgfouca I need to rebase this branch to get the latest changes for cheyenne - is that okay with you?

**Comment by jgfouca** [comment]
> - is that okay with you?

yes

**Comment by jedwards4b** [comment]
@jgfouca I think this is just about there but it requires minor changes in each component buildlib.  I've done all of the cesm ones - how do you want to handle the e3sm ones?

**Comment by jgfouca** [comment]
@jedwards4b , what kinds of changes were needed?

**Comment by jedwards4b** [comment]
This is typical:
```
diff --git a/cime_config/buildlib b/cime_config/buildlib
index 04fe207..a668f72 100755
--- a/cime_config/buildlib
+++ b/cime_config/buildlib
@@ -11,6 +11,7 @@ from standard_script_setup import *
 from CIME.case import Case
 from CIME.utils import expect, run_cmd
 from CIME.buildlib import parse_input
+from CIME.build import get_standard_makefile_args
 
 logger = logging.getLogger(__name__)
 
@@ -31,7 +32,7 @@ def _build_mosart():
         if not os.path.isfile(filepath_file):
             srcroot = case.get_value("SRCROOT")
             caseroot = case.get_value("CASEROOT")
-            paths = [ os.path.join(caseroot,"SourceMods","src.mosart"), 
+            paths = [ os.path.join(caseroot,"SourceMods","src.mosart"),
                       os.path.join(srcroot,"components","mosart","src","riverroute"),
                       os.path.join(srcroot,"components","mosart","src","cpl")]
 
@@ -43,8 +44,8 @@ def _build_mosart():
         complib = os.path.join(libroot,"librof.a")
         makefile = os.path.join(casetools, "Makefile")
 
-        cmd = "%s complib -j %d MODEL=mosart COMPLIB=%s -f %s" \
-            % (gmake, gmake_j, complib, makefile)
+        cmd = "{} complib -j {} MODEL=mosart COMPLIB={} -f {} {}" \
+            .format(gmake, gmake_j, complib, makefile, get_standard_makefile_args(case))
 
         rc, out, err = run_cmd(cmd, from_dir=bldroot)
         expect(rc == 0, "Command %s failed rc=%d\nout=%s\nerr=%s" % (cmd, rc, out, err))
@@ -55,6 +56,3 @@ def _build_mosart():
 
 if __name__ == "__main__":
     _build_mosart()
```
I also added a cli that you can call if you still have buildlib in perl

**Comment by jgfouca** [comment]
I'll worry about the E3SM buildlibs when I merge CIME.

**Pull request merged**